### PR TITLE
ci(bump automation): bump ubi9 for ironbank

### PR DESCRIPTION
**Pull request by v1v**
### What

Enable updatecli [policies](https://www.updatecli.io/blog/shareable-policies) to bump the Ironbank versions automatically, then 
https://github.com/elastic/elasticsearch/pull/111743 won't be manually created but when a new ubit9 version is released and available in the Ironbank system.

Those policies can be found at [elastic/oblt-updatecli-policies@main/updatecli/policies/](https://github.com/elastic/oblt-updatecli-policies/tree/main/updatecli/policies/?rgh-link-date=2024-08-28T16%3A30%3A04Z) (NOTE: This is a private repository only accessible by Elastic employees)

#### How to test this PR locally

1. `gh pr checkout 112298`
2. Install [updatecli](https://www.updatecli.io/docs/prologue/installation/)
3. Login to [ghcr.io](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry)
4. Diff (dry-run)
```bash
$ GITHUB_TOKEN=$(gh auth token) updatecli compose diff --experimental
```
5. Create Pull Request if new changes
```bash
$ GITHUB_REPOSITORY=elastic/elasticsearch \
   GITHUB_ACTOR=v1v \
   GITHUB_TOKEN=$(gh auth token) \
updatecli compose apply --experimental
```

### CODE REVIEW

**Comment by mark-vieira** [comment]
@v1v how will this deal with the fact that the version also needs to be updated in the hardening manifest file as well? See https://github.com/elastic/elasticsearch/pull/112329/files.

**Comment by v1v** [comment]
> @v1v how will this deal with the fact that the version also needs to be updated in the hardening manifest file as well? See [#112329 (files)](https://github.com/elastic/elasticsearch/pull/112329/files).

Good question; it will do it automatically and update both files (`Dockerfile` and `hardening_manifest.yaml`) in the given path. That's the default behaviour.

In this case, if the Dockerfile name is not Dockerfile or hosted on the same path, then it's possible to change the location (relative to the path) and the location. The same happens for the `hardening_manifest.yaml`.

In other words:

```yaml
config:
  - path: distribution/docker/src/docker/iron_bank
    dockerfile: ../Dockerfile
    manifest: hardening_manifest.yaml
```

is the explicit configuration for this PR. If it helps, I can change it for sure.


A bit more context: That's how it was initially implemented for https://github.com/elastic/apm-server/pull/13934/files#diff-16cc21c0ab87247dbc828d81e4c8c5fcec44f535ec6e1deab14a36e86a1a6002 and https://github.com/elastic/kibana/pull/191660, but it seems there are some projects using ironbark at elastic with templates or different bits and pieces, see https://github.com/elastic/beats/pull/40633/files#diff-0572bf311a603d91d349a6c3adcfab47a70f85d48390e24a52600951b97ef77e and https://github.com/elastic/ent-search/pull/8086/files#diff-96892c1c22309fa1b67a9cf0acf05b8354ca7d5128815ab5b336d709723750a1

Let me know your thoughts and if the configuration syntax is clear, we can make any changes as we speak. 

Thanks for your feedback Mark

**Comment by mark-vieira** [comment]
One more question @v1v: how does this tool overlap with Renovate which is being used for automated updates for other dependencies, like Chainguard images. Is there a reason we have two separate tools here?

**Review by mark-vieira — APPROVED** [review]

**Comment by v1v** [comment]
> how does this tool overlap with Renovate which is being used for automated updates for other dependencies, like Chainguard images. Is there a reason we have two separate tools here?

Renovate will eventually support it; I have already requested support from the Platform team for Ironbank with Renovate.  I'm just waiting for the implementation.

While waiting I decided to avoid manually updating those files for the `APM Server` and automate the process with `updatecli`, a tool we use for many things that require changing files from vast, diverse sources. Hence, I created a shared updatecli policy that can be used by some other projects, such as Elasticsearch, Kibana, Elastic Agent, Beats and Entsearch - I discovered recently that some files were not correctly updated or were using an old version in some of the projects mentioned above.

It's up to the teams to decide whether to use this approach or wait for support in Renovate.

**Comment by mark-vieira** [comment]
Awesome. Thanks @v1v!

File: .github/updatecli/values.d/scm.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  branch: main
+  # begin updatecli-compose policy values
+  user: elasticmachine
+  email: 42973632+elasticmachine@users.noreply.github.com
+  commitusingapi: true
+  # end updatecli-compose policy values
+
+# This will be moved to the scm section in the future and use
+# commitusingapi instead.
+signedcommit: true
```
Comment (v1v) [inline .github/updatecli/values.d/scm.yml:14]:
```suggestion
  commitusingapi: true
  # begin updatecli-compose policy values
  user: elasticmachine
  email: 42973632+elasticmachine@users.noreply.github.com
  # end updatecli-compose policy values
```

File: updatecli-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Config file for `updatecli compose ...`.
+# https://www.updatecli.io/docs/core/compose/
+policies:
+  - name: Handle ironbank bumps
+    policy: ghcr.io/elastic/oblt-updatecli-policies/ironbank/templates:0.1.0@sha256:00483be31c74d3f95ea37d1ca7aeee98131ab2c5c1f901a043e9afb122a445b0
```
Comment (v1v) [inline updatecli-compose.yaml:5]:
```suggestion
    policy: ghcr.io/elastic/oblt-updatecli-policies/ironbank/templates:0.3.0@sha256:b0c841d8fb294e6b58359462afbc83070dca375ac5dd0c5216c8926872a98bb1
```

**Review by reakaleek — APPROVED** [review]

**Comment by v1v** [comment]
@mark-vieira , do you know what's the reason for the failure in the CI? I tried a few reruns but same behaviour. Do I need to rebase my PR?

**Comment by v1v** [comment]
I'll rebase my changes and see if the CI passes... however, the CI does not test these changes

**Pull request merged**